pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -732,58 +749,49 @@ def tickSpacing(self, minVal, maxVal, size):
         dif = abs(maxVal - minVal)
         if dif == 0:
             return []
-
-        ## decide optimal minor tick spacing in pixels (this is just aesthetics)
-        optimalTickCount = max(2., log(size))
-
-        ## optimal minor tick spacing
-        optimalSpacing = dif / optimalTickCount
-
-        ## the largest power-of-10 spacing which is smaller than optimal
-        p10unit = 10 ** floor(log10(optimalSpacing))
-
-        ## Determine major/minor tick spacings which flank the optimal spacing.
-        intervals = np.array([1., 2., 10., 20., 100.]) * p10unit
-        minorIndex = 0
-        while intervals[minorIndex+1] <= optimalSpacing:
-            minorIndex += 1
-
+        
+        ref_size = 300. # axes longer than this display more than the minimum number of major ticks
+        minNumberOfIntervals = max(
+            2.25,       # 2.0 ensures two tick marks. Fudged increase to 2.25 allows room for tick labels. 
+            2.25 * self._tickDensity * sqrt(size/ref_size) # sub-linear growth of tick spacing with size
+        )
+
+        majorMaxSpacing = dif / minNumberOfIntervals        
+        p10unit = 10 ** floor(log10(majorMaxSpacing))  
```

Comment:
feels like there is a performance speedup to be had here with utilizing `int(majorMaxSpacing).bit_length()`, avoiding the log calculation outright; but I'm a bit tired to pencil it out to see what the equivalent would be (given that gives you a value in log_2_.

Reply:
Well, the float also knows its own base 2 exponent already.

Reply:
good point!